crypto/src/main/java/org/web3j/crypto/StructuredDataEncoder.java

```diff
@@ -88,14 +88,15 @@ public Set<String> getDependencies(String primaryType) {
             remainingTypes.remove(remainingTypes.size() - 1);
             deps.add(structName);
 
-            for (StructuredData.Entry entry : types.get(primaryType)) {
-                if (!types.containsKey(entry.getType())) {
+            for (StructuredData.Entry entry : types.get(structName)) {
+                String baseDeclarationTypeName = getBaseTypeNameFromDeclaration(entry.getType());
```

Comment:
```suggestion
                String baseDeclarationTypeName = arrayTypePattern.matcher(declaration).find() ? declaration.substring(0, declaration.indexOf('[')) : declaration;
```
no additional method need it

Reply:
ok, I have apply and fix and test the suggested code

Reply:
please run :spotlessJavaApply in order to solve code  style issue